oozie-to-airflow/converter/primitives.py

```diff
@@ -14,53 +14,72 @@
 # limitations under the License.
 """Class for Airflow relation"""
 from collections import OrderedDict
-from typing import Set, Optional, Dict, NamedTuple, Any
+from typing import Any, Dict, Optional, Set
+
+from orderedset import OrderedSet
 
 from airflow.utils.trigger_rule import TriggerRule
 
-# Pylint and flake8 does not understand forward references
-# https://www.python.org/dev/peps/pep-0484/#forward-references
-from converter import parsed_node  # noqa: F401 pylint: disable=unused-import
 from utils.template_utils import render_template
+from utils.variable_name import convert_to_python_variable
 
 
-class Relation(NamedTuple):
+class Relation:
     """Class for Airflow relation"""
 
-    from_task_id: str
-    to_task_id: str
+    def __init__(self, from_task_id: str, to_task_id: str):
```

Comment:
Why can not it be a tuple?  IMO, This is only needed when you overwrite the constructor.

Reply:
Because if you want to use "@property" mypy complains.

Reply:
https://github.com/GoogleCloudPlatform/cloud-composer/commits/test-mypy
It works for me.